File: src/main/java/org/projetperso/expense_tracking/model/Expense.java
package org.projetperso.expense_tracking.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
public class Expense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String description;

    @NotNull
    @Positive
    private BigDecimal amount;

    @NotNull
    private LocalDate date;

    @NotNull
    @Enumerated(EnumType.STRING)
    private Category category;

    // Getters
    public Long getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public LocalDate getDate() {
        return date;
    }

    public Category getCategory() {
        return category;
    }

    // Setters
    public void setId(Long id) {
        this.id = id;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public enum Category {
        FOOD,
        TRANSPORT,
        HOUSING,
        UTILITIES,
        ENTERTAINMENT,
        OTHER
    }
}

File: expense-tracking-frontend/src/types/expense.ts
export interface ExpenseFormData {
    description: string;
    amount: number;
    date: string;
    category: string;
}

export enum ExpenseCategory {
    FOOD = 'FOOD',
    HOUSING = 'HOUSING',
    TRANSPORT = 'TRANSPORT',
    UTILITIES = 'UTILITIES',
    ENTERTAINMENT = 'ENTERTAINMENT',
    OTHER = 'OTHER'
}

export interface ExpenseFormData {
    description: string;
    amount: number;
    date: string;
    category: ExpenseCategory; // Maintenant nous utilisons l'enum
}

export interface Expense extends ExpenseFormData {
    id: number;
}

